From 8fbc303f1bb9e9d4f101d4af69669a01ce4ed359 Mon Sep 17 00:00:00 2001
From: gkdeepa <g.k.deepa@intel.com>
Date: Tue, 24 Aug 2021 08:04:00 +0530
Subject: [PATCH] Include VHAL for camera

legacy hal inclusion
properties declaration
product package inclusion for libs

Tracked-On:OAM-99722
Signed-off-by: gkdeepa g.k.deepa@intel.com
---
 groups/camera-ext/ext-camera-only/product.mk | 17 +++++++++++++++--
 groups/device-specific/caas_cfc/manifest.xml |  1 +
 2 files changed, 16 insertions(+), 2 deletions(-)

diff --git a/groups/camera-ext/ext-camera-only/product.mk b/groups/camera-ext/ext-camera-only/product.mk
index 815464c..84cb4ad 100644
--- a/groups/camera-ext/ext-camera-only/product.mk
+++ b/groups/camera-ext/ext-camera-only/product.mk
@@ -5,9 +5,22 @@ PRODUCT_COPY_FILES += \
 
 # External camera service
 PRODUCT_PACKAGES += android.hardware.camera.provider@2.4-external-service \
+                    android.hardware.camera.provider@2.4-service_64 \
                     android.hardware.camera.provider@2.4-impl
+#VHAL camera
+PRODUCT_PACKAGES += camera.$(TARGET_BOARD_PLATFORM) \
+                    camera.$(TARGET_BOARD_PLATFORM).jpeg
 
+PRODUCT_PROPERTY_OVERRIDES += ro.vendor.remote.sf.fake_camera ="both" \
+                              ro.vendor.camera.in_frame_format.h264=false \
+                              ro.vendor.camera.in_frame_format.i420=true \
+                              ro.vendor.camera.decode.vaapi=false \
+                              ro.vendor.remote.sf.back_camera_hal= \
+                              ro.vendor.remote.sf.front_camera_hal= \
+                              ro.vendor.camera.transference="VSOCK" \
+                              vendor.camera.external="VHAL"
+#removing not required apps
 # Only include test apps in eng or userdebug builds.
-PRODUCT_PACKAGES_DEBUG += TestingCamera
+#PRODUCT_PACKAGES_DEBUG += TestingCamera
 
-PRODUCT_PACKAGES += MultiCameraApp
+#PRODUCT_PACKAGES += MultiCameraApp
diff --git a/groups/device-specific/caas_cfc/manifest.xml b/groups/device-specific/caas_cfc/manifest.xml
index 499945f..5a41ab8 100644
--- a/groups/device-specific/caas_cfc/manifest.xml
+++ b/groups/device-specific/caas_cfc/manifest.xml
@@ -149,6 +149,7 @@
         <interface>
             <name>ICameraProvider</name>
             <instance>external/0</instance>
+            <instance>legacy/0</instance>
         </interface>
     </hal>
     <hal format="hidl">
-- 
2.17.1

